client/app/services/query.js

```diff
@@ -83,14 +164,51 @@ export class Parameter {
     return isNull(this.getValue());
   }
 
+  get hasDynamicDate() {
+    return isDateParameter(this.type) && isDynamicDate(this.value);
+  }
+
+  get dynamicDate() {
+    return getDynamicDate(this.value);
+  }
+
+  get hasDynamicDateRange() {
+    return isDateRangeParameter(this.type) && isDynamicDateRange(this.value);
+  }
+
+  get dynamicDateRange() {
+    return getDynamicDateRange(this.value);
+  }
+
```

Comment:
Is there a way around having type specific code in the generic param object?
Can this be reduced to `get hasDynamicValue()` and `get dynamicValue()`? (leaving out data type)

Reply:
> Is there a way around having type specific code in the generic param object?

Inheritance! :tada: 

> Can this be reduced to get hasDynamicValue() and get dynamicValue()? (leaving out data type)

That was my first approach, I'm not so sure why I changed this part too... I'll move it back to it in this part and in case it makes sense I separate it back in the future when parameters are destructured .